src/components/HomeUpload/HomeUpload.js

```diff
@@ -11,8 +11,33 @@ import AppContext from "../../AppContext";
 export default function HomeUpload() {
   const [files, setFiles] = useState([]);
   const { apiUrl } = useContext(AppContext);
+  const [directoryMode, setDirectoryMode] = useState(false);
+
+  useEffect(() => {
+    if (directoryMode) {
+      inputRef.current.setAttribute("webkitdirectory", "true");
```

Comment:
Is there a race condition where `inputRef.current` is undefined?

Reply:
no, the ref is assigned before the useEffect is ran